Production/python/customiseHLT.py

```diff
@@ -0,0 +1,147 @@
+# How to run:
+# hltGetConfiguration /dev/CMSSW_12_4_0/GRun --globaltag auto:phase1_2022_realistic --mc --unprescale --no-output --max-events 100 --eras Run3 --l1-emulator FullMC --l1 L1Menu_Collisions2022_v1_3_0-d1_xml --customise TauMLTools/Production/customiseHLT.customise --input /store/mc/Run3Summer21DRPremix/TT_TuneCP5_14TeV-powheg-pythia8/GEN-SIM-DIGI-RAW/120X_mcRun3_2021_realistic_v6-v2/2540000/b354245e-d8bc-424d-b527-58815586a6a5.root > hltRun3Summer21MC.py
+# cmsRun hltRun3Summer21MC.py
+# file dataset=/store/mc/Run3Summer21DRPremix/TT_TuneCP5_14TeV-powheg-pythia8/GEN-SIM-DIGI-RAW/120X_mcRun3_2021_realistic_v6-v2/2540000/00c642d1-bf7e-477d-91e1-4dd9ce2c8099.root
+import FWCore.ParameterSet.Config as cms
+from PhysicsTools.NanoAOD.common_cff import Var, P4Vars
+
+def customiseGenParticles(process):
+  def pdgOR(pdgs):
+    abs_pdgs = [ f'abs(pdgId) == {pdg}' for pdg in pdgs ]
+    return '( ' + ' || '.join(abs_pdgs) + ' )'
+
+  leptons = pdgOR([ 11, 13, 15 ])
+  important_particles = pdgOR([ 6, 23, 24, 25, 35, 39, 9990012, 9900012 ])
```

Comment:
@kandrosov Could you comment on this question?

Reply:
- 9990012/9900012 are HNL (Majorana and Dirac). We'll use HNL samples to train on displaced taus. Perhaps we can also add stau to the list. @shedprog what pdg ids staus have in our MC?
- genJets will be stored as a separate collection equivalent to the `GenJet` nano collection. So if tau doesn't match to any lepton, but matches to genJet - it will be labeled as jet faking tau.

Reply:
In the maximally mixed scenario that we are considering, it is +1000015/-1000015, in other scenarios it might be the number with 7 numbers always ends on *15